exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -1500,3 +1503,63 @@ func (ku *Kucoin) ValidateAPICredentials(ctx context.Context, assetType asset.It
 	_, err := ku.UpdateAccountInfo(ctx, assetType)
 	return ku.CheckTransientError(err)
 }
+
+// GetFuturesContractDetails returns details about futures contracts
+func (ku *Kucoin) GetFuturesContractDetails(ctx context.Context, item asset.Item) ([]futures.Contract, error) {
+	if !item.IsFutures() {
+		return nil, futures.ErrNotFuturesAsset
+	}
+	if !ku.SupportsAsset(item) || item != asset.Futures {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, item)
+	}
+
+	contracts, err := ku.GetFuturesOpenContracts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := make([]futures.Contract, len(contracts))
+	for i := range contracts {
+		var cp, underlying currency.Pair
+		underlying, err = currency.NewPairFromStrings(contracts[i].BaseCurrency, contracts[i].QuoteCurrency)
+		if err != nil {
+			return nil, err
+		}
+		cp, err = currency.NewPairFromStrings(contracts[i].BaseCurrency, contracts[i].Symbol[len(contracts[i].BaseCurrency):])
+		if err != nil {
+			return nil, err
+		}
+		settleCurr := currency.NewCode(contracts[i].SettleCurrency)
+		var ct futures.ContractType
+		if contracts[i].ContractType == "FFWCSX" {
+			ct = futures.Perpetual
+		} else {
+			ct = futures.Quarterly
+		}
+		direction := futures.Linear
+		if contracts[i].IsInverse {
+			direction = futures.Inverse
+		}
+		resp[i] = futures.Contract{
+			Exchange:             ku.Name,
+			Name:                 cp,
+			Underlying:           underlying,
+			SettlementCurrencies: currency.Currencies{settleCurr},
+			MarginCurrency:       settleCurr,
+			Asset:                item,
+			StartDate:            contracts[i].FirstOpenDate.Time(),
+			EndDate:              contracts[i].ExpireDate.Time(),
+			IsActive:             !strings.EqualFold(contracts[i].Status, "closed"),
+			Status:               contracts[i].Status,
+			Multiplier:           contracts[i].Multiplier,
+			MaxLeverage:          contracts[i].MaxLeverage,
+			Direction:            direction,
+			LatestRate: fundingrate.Rate{
+				Rate: decimal.NewFromFloat(contracts[i].FundingFeeRate),
+				Time: time.Unix(contracts[i].NextFundingRateTime, 0).Add(-time.Hour * 8), // kucoin contracts every 8 hours
```

Comment:
For the life of me I have no idea what they are sending back in time terms when testing this.

example:
```
"nextFundingRateTime":9405779,
```

All the rates end up like this:

```
 LatestRate:{Time:1970-04-19 23:01:47 +1000 AEST Rate:0.0001 Payment:0}
```

Sorry I cant be more helpful. Maybe they are millseconds until next funding rate?

Reply:
Looks like it's the delta between time.Now and the next settlement time in milliseconds

Reply:
Sweet catch! Yeah its ms until the next one